config/software/datadog-agent-integrations.rb

```diff
@@ -89,29 +98,34 @@
     if windows?
       command("#{python_bin} -m #{python_pip_no_deps}\\datadog_checks_base")
       command("#{python_bin} -m piptools compile --generate-hashes --output-file #{windows_safe_path(project_dir)}\\static_requirements.txt #{windows_safe_path(project_dir)}\\datadog_checks_base\\datadog_checks\\data\\agent_requirements.in")
-
-      # Uninstall the deps that pip-compile installs so we don't include them in the final artifact
-      for dep in UNINSTALL_PIPTOOLS_DEPS
-        command("#{python_bin} -m #{python_pip_uninstall} #{dep}")
-      end
-
-      command("#{python_bin} -m #{python_pip_reqs} #{windows_safe_path(project_dir)}\\static_requirements.txt")
     else
       build_env = {
         "LD_RUN_PATH" => "#{install_dir}/embedded/lib",
         "PATH" => "#{install_dir}/embedded/bin:#{ENV['PATH']}",
       }
       pip "install --no-deps .", :env => build_env, :cwd => "#{project_dir}/datadog_checks_base"
       command("#{install_dir}/embedded/bin/python -m piptools compile --generate-hashes --output-file #{project_dir}/static_requirements.txt #{project_dir}/datadog_checks_base/datadog_checks/data/agent_requirements.in")
+    end
 
-      # Uninstall the deps that pip-compile installs so we don't include them in the final artifact
-      for dep in UNINSTALL_PIPTOOLS_DEPS
-        pip "uninstall -y #{dep}"
+    # Uninstall the deps that pip-compile installs so we don't include them in the final artifact
+    UNINSTALL_PIPTOOLS_DEPS.each do |dep|
+      if not frozen_agent_reqs.include? "#{dep}=="
```

Comment:
If pip-tools includes `dummy==1.0` and we want to ship `toto-dummy==2.6`, `toto-dummy==2.6` will get removed

Reply:
No, that's not accurate. We'd *NOT* remove it, which is a lesser problem, but I'll change this for  a regex match.

Reply:
Oh yeah you are right I don't know why but I assumed that we were uninstalling what was matched and not `dep`...
Sorry for the noise. I think we should still fix it in case we modify this logic in the future.
I would `include` to 
https://www.rubydoc.info/stdlib/core/1.9.2/String#start_with%3F-instance_method